### PR TITLE
adding the data loader function and the data load function test noteb…

### PR DESCRIPTION
**Pull request by mjw-18**
…ook. there was also a very minor change to the sample_info_data_visualizer notebook. All edited notebooks have had their changes converted to their appropriate modules nbconverted folder

### CODE REVIEW

**Comment by mjw-18** [comment]
hey @gwaybio I believe that this is ready for review!

File: 0.data-download/scripts/data_loader.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    # Load data
+    dependency_df = pd.read_csv(dependency_data_file, index_col=0).reset_index()
+    sample_df = pd.read_csv(sample_info_file, index_col=0)
+    sample_df = sample_df.set_index("DepMap_ID").reset_index()
+
+    # searching for similar IDs FROM dependency df IN sample df
+    dep_ids = dependency_df["DepMap_ID"].tolist()
+    dep_vs_samp_ids = set(dep_ids) & set(sample_df["DepMap_ID"].tolist())
+
+    # searching for similar IDs FROM s ample df In dependency df
```
Comment (gwaybio) [inline 0.data-download/scripts/data_loader.py]:
```suggestion
    # searching for similar IDs FROM sample df In dependency df
```

File: 0.data-download/scripts/data_loader.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    samp_ids = sample_df["DepMap_ID"].tolist()
+    samp_vs_dep_ids = set(samp_ids) & set(sample_df["DepMap_ID"].tolist())
+    
+    # extract matching IDs
```
Comment (gwaybio) [inline 0.data-download/scripts/data_loader.py]:
```suggestion
    # subset data to only matching IDs (samples in both dependency and sample data)
```

File: 0.data-download/scripts/data_loader.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    if adult_or_pediatric == "all":
+       dependency_df = dependency_df
+       samples_to_keep = sample_df.reset_index(drop=True).DepMap_ID.tolist()
+       sample_df = sample_df.query("DepMap_ID == @samples_to_keep").reset_index(drop=True)
+    elif adult_or_pediatric != "all":
+       sample_df = sample_df.query("age_categories == @adult_or_pediatric").reset_index(drop=True)
+       samples_to_keep = sample_df.reset_index(drop=True).DepMap_ID.tolist()  
+       dependency_df = dependency_df.query("DepMap_ID == @samples_to_keep").reset_index(drop=True)
+    return sample_df, dependency_df
```
Comment (gwaybio) [inline 0.data-download/scripts/data_loader.py]:
```suggestion
    if adult_or_pediatric != "all":
       sample_df = sample_df.query("age_categories == @adult_or_pediatric").reset_index(drop=True)
       samples_to_keep = sample_df.reset_index(drop=True).DepMap_ID.tolist()  
       dependency_df = dependency_df.query("DepMap_ID == @samples_to_keep").reset_index(drop=True)
    
    return sample_df, dependency_df
```

I think all the logic in `if adult_or_pediatric == "all"` is redundant, so it is best to skip. What do you think?

**Review by gwaybio — CHANGES_REQUESTED** [review]
A couple changes to consider.

Also, I'd keep `test_data_loader.ipynb` on your local computer, no need to add here. You can add this to gitignore

**Pull request merged**